.vscode/tasks.json

```
@@ -1,58 +1,43 @@
+
 {
   "tasks": [
     {
+      "label": "WaitForDebugpy",
       "type": "shell",
-      "label": "conda: activate environment",
-      "command": "activate freecad",
+      "command": "python ${workspaceFolder}/.vscode/WaitForPort.py",
+      "group": "none",
       "problemMatcher": [],
-      "detail": "Activate conda environment",
-      "group": "build"
+      "presentation": {
+        "reveal": "never", //silently fail and don't launch the debugger
+        "panel": "dedicated",
+        "close": true,
+        "revealProblems": "never"
+      },
+      "hide": true
     },
     {
-      "type": "cmake",
+      "type": "shell",
       "label": "CMake: configure",
-      "command": "configure",
+      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { activate freecad };             cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
+      "linux":   { "command": "if echo \"${command:cmake.activeConfigurePresetName}\" | grep -iqF \"conda\"; then source activate freecad; fi; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
+      "osx":     { "command": "if echo \"${command:cmake.activeConfigurePresetName}\" | grep -iqF \"conda\"; then source activate freecad; fi; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
       "problemMatcher": [],
-      "detail": "CMake template configure task",
-      "group": "build"
+      "detail": "Configure the project"
     },
     {
-      "type": "cmake",
+      "type": "shell",
       "label": "CMake: build",
-      "command": "build",
-      "targets": ["all"],
+      //"windows": { "command": "IF /I \"${command:cmake.activeConfigurePresetName}\" == \"conda\" activate freecad;                             cmake --build ${command:cmake.buildDirectory} --parallel" },
+      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { activate freecad };             cmake --build ${command:cmake.buildDirectory} --parallel" },
```

similar to the mentioned above:
```suggestion
     "windows": { "command": "${workspaceFolder}/.vscode/scripts/cmake-build.bat '${command:cmake.buildDirectory}' '${command:cmake.activeConfigurePresetName}'" },
```
and `cmake-build.bat`:
```cmd
echo %2 | find /i "conda" >nul && (
    call .conda\freecad\condabin\activate.bat
)

cmake --build %1 --parallel
```